src/Altinn.App.Core/Models/Validation/ValidationIssue.cs

```diff
@@ -71,7 +71,10 @@ public class ValidationIssue
         /// </remarks>
         [JsonProperty(PropertyName = "source")]
         [JsonPropertyName("source")]
-        public string Source { get; set; } = default!;
+#nullable disable
+        public string Source { get; set; }
+
+#nullable restore
```

Comment:
See `<remarks>`:

```suggestion
        public string Source { get; set; } = "[]"
```

We need a permanent solution for this. It can't be required, but it makes no sense if marked as nullable, as we set it for all types before returning.

Reply:
Did read the remark, but still opted for keeping the same behavior. In the second to last commit I implemented what you suggested and added some test assertions to verify. Does that look OK to you?

Reply:
Actually, I remember now I searched for `new ValidationIssue` and found that `Source` is not at all always non-null. So I whole-heartedly regret that commit 😄 

Here is one example
![image](https://github.com/Altinn/app-lib-dotnet/assets/5425986/6edabd09-7fa5-478a-ae50-38b677a1f138)
So now `Source` will be `"[]"` as opposed to null. I don't think this makes any sense. This is a classic case where the warning that was emitted should have led to a better design as opposed to being ignored/suppressed IMO